### PR TITLE
Updates to release checklist.

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

Adding items that showed up in the recent release that were not written down or required better clarification.

**Which issue(s) this PR fixes**:

N/A

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
@mangalpalli - wanted to get your review on this, since you follow the steps. Does this make sense?

File: docs/governance/templates/release_issue.md [hunk 1]
```diff
@@ -12,8 +12,11 @@ and copy it into a release issue. Fill in relevant values, found inside {}
 - [ ] Have at least `Editor` level access to `agones-images` project.
 - [ ] Have permission to publish new versions of the App Engine application.
 - [ ] Have write access to Agones github repository.
-- [ ] Have a `gcloud config configurations` configuration called `agones-images` that points to the same project.
+- [ ] Run `git remote -v` and see"
```
Comment (roberthbailey) [inline docs/governance/templates/release_issue.md:15]: trailing `"`

File: docs/governance/templates/release_issue.md [hunk 2]
```diff
@@ -73,6 +75,7 @@ and copy it into a release issue. Fill in relevant values, found inside {}
     (`helm install --namespace=agones-system agones agones/agones --version={version}-rc` if release candidate) and a smoke test to confirm everything is working.
 - [ ] Attach all assets found in the `release` folder to the draft Github Release.
 - [ ] If release candidate check the pre-release box on the draft Github Release
+- [ ] Copy any review changes to the release blog post are copied into the draft Github release.
```
Comment (roberthbailey) [inline docs/governance/templates/release_issue.md:78]: "Ensure that any review changes ..." or "Copy any review changes into the draft ..."

**Comment by mangalpalli** [comment]
> @mangalpalli - wanted to get your review on this, since you follow the steps. Does this make sense?

I've reviewed, looks good to me.

File: docs/governance/templates/release_issue.md (hunk 2 above)
Comment (markmandel) [reply docs/governance/templates/release_issue.md]: Ermagerd that sentence makes no sense. Thanks.

**Comment by markmandel** [comment]
That's a new flake in `push-images`

```
#6 [2/4] RUN apk --update add ca-certificates &&     adduser -D -u 1000 agones
#6 CACHED

#7 [3/4] COPY --chown=agones:agones ./bin/sdk-server.linux.arm64 /home/agones/sdk-server
#7 ERROR: failed to calculate checksum of ref 6223a7xilmc6foa1yuhvg2qx5::nkrn9x5gut7gehsaoc64un0mz: "/bin/sdk-server.linux.arm64": not found
------
 > [3/4] COPY --chown=agones:agones ./bin/sdk-server.linux.arm64 /home/agones/sdk-server:
------
Dockerfile:20
--------------------
  18 |         adduser -D -u 1000 agones
  19 |     
  20 | >>> COPY --chown=agones:agones ./bin/sdk-server.linux.$ARCH /home/agones/sdk-server
  21 |     COPY --chown=agones:agones ./bin/LICENSES ./bin/dependencies-src.tgz /home/agones/
  22 |     
--------------------
error: failed to solve: failed to compute cache key: failed to calculate checksum of ref 6223a7xilmc6foa1yuhvg2qx5::nkrn9x5gut7gehsaoc64un0mz: "/bin/sdk-server.linux.arm64": not found
make[1]: *** [Makefile:495: build-agones-sdk-image-arm64] Error 1
make[1]: Leaving directory '/workspace/build'
make: *** [Makefile:607: push-agones-sdk-linux-image-arm64] Error 2
```

**Comment by roberthbailey** [comment]
It failed in `push-images`:

```
error: failed to solve: alpine:3.14: failed to do request: Head "https://registry-1.docker.io/v2/library/alpine/manifests/3.14": EOF
```

I've verified that it works for me locally:

```
$ docker pull alpine:3.14
3.14: Pulling from library/alpine
c7ed990a2339: Pull complete 
Digest: sha256:1ab24b3b99320975cca71716a7475a65d263d0b6b604d9d14ce08f7a3f67595c
Status: Downloaded newer image for alpine:3.14
docker.io/library/alpine:3.14
```
So this looks like a flake.

**Review by roberthbailey — APPROVED** [review]

**Pull request merged**